### ISSUE TITLE
No telemetry reported for External ServiceEntry with Gateways

### ISSUE THREAD

**Issue by josejulio** [istio/istio#15188]
**Bug description**

I was following [Perform TLS origination with an egress gateway](https://istio.io/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#perform-tls-origination-with-an-egress-gateway) example and found out that Telemetry to the service entry is not being reported when doing the Step 5.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

ServiceEntry telemetry is being reported e.g. time-serie for `istio_tcp_sent_bytes_total{destination_service="edition.cnn.com"}` is updated.

**Steps to reproduce the bug**

1. Follow [Perform TLS origination with an egress gateway ](https://istio.io/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#perform-tls-origination-with-an-egress-gateway)example
  - Skip step 2 of the example, at that point, it produces the right telemetry, and could cause confusion when checking later.
  - It's the same with mutual tls enabled or disabled (step 3)
2. After completing, notice that further calls to the ServiceEntry (e.g. step 5) no longer generates any telemetry for `istio_tcp_sent_bytes_total{destination_service="edition.cnn.com"}` or `istio_requests_total{destination_service="edition.cnn.com"}`

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

```
$ istioctl version --remote
client version: 1.2.0
citadel version: 1.2.0
egressgateway version: 1.2.0
galley version: 1.2.0
ingressgateway version: 1.2.0
pilot version: 1.2.0
policy version: 1.2.0
sidecar-injector version: 1.2.0
telemetry version: 1.2.0
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.7", GitCommit:"0c38c362511b20a098d7cd855f1314dad92c2780", GitTreeState:"clean", BuildDate:"2018-08-20T10:09:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-06-20T16:29:27Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

Installing istio with helm

**Environment where bug was observed (cloud vendor, OS, etc)**

Fedora 29

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by fpesce** [comment]
Reproduced on 1.1.14 / 1.2.5 and 1.3.0-rc2 
@douglas-reid is working on a fix.

**Comment by douglas-reid** [comment]
I believe this to be fixed in 1.4.0-beta.2. `edition.cnn.com` gets upgraded to http/2, but I see the following, via prometheus:

*sleep -> egress gw*
`istio_requests_total{connection_security_policy="unknown",destination_app="istio-egressgateway",destination_principal="unknown",destination_service="istio-egressgateway.istio-system.svc.cluster.local",destination_service_name="istio-egressgateway",destination_service_namespace="istio-system",destination_version="unknown",destination_workload="istio-egressgateway",destination_workload_namespace="istio-system",instance="10.56.1.11:42422",job="istio-mesh",permissive_response_code="none",permissive_response_policyid="none",reporter="source",request_protocol="http",response_code="301",response_flags="-",source_app="sleep",source_principal="unknown",source_version="unknown",source_workload="sleep",source_workload_namespace="default"} | 1`

*egress gw -> cnn*
`istio_requests_total{connection_security_policy="unknown",destination_app="unknown",destination_principal="unknown",destination_service="edition.cnn.com",destination_service_name="edition.cnn.com",destination_service_namespace="default",destination_version="unknown",destination_workload="unknown",destination_workload_namespace="unknown",instance="10.56.1.11:42422",job="istio-mesh",permissive_response_code="none",permissive_response_policyid="none",reporter="source",request_protocol="http",response_code="301",response_flags="-",source_app="istio-egressgateway",source_principal="unknown",source_version="unknown",source_workload="istio-egressgateway",source_workload_namespace="istio-system"}`

TCP: *sleep -> egress gw*
`istio_tcp_sent_bytes_total{connection_security_policy="unknown",destination_app="unknown",destination_principal="unknown",destination_service="edition.cnn.com",destination_service_name="edition.cnn.com",destination_service_namespace="default",destination_version="unknown",destination_workload="unknown",destination_workload_namespace="unknown",instance="10.56.1.11:42422",job="istio-mesh",reporter="source",response_flags="-",source_app="sleep",source_principal="unknown",source_version="unknown",source_workload="sleep",source_workload_namespace="default"}	1312420`

More confirmation is needed for plain HTTP traffic.

**Comment by douglas-reid** [comment]
For plain HTTP:

*sleep -> egress gw*
`istio_requests_total{connection_security_policy="unknown",destination_app="istio-egressgateway",destination_principal="unknown",destination_service="istio-egressgateway.istio-system.svc.cluster.local",destination_service_name="istio-egressgateway",destination_service_namespace="istio-system",destination_version="unknown",destination_workload="istio-egressgateway",destination_workload_namespace="istio-system",instance="10.56.1.11:42422",job="istio-mesh",permissive_response_code="none",permissive_response_policyid="none",reporter="source",request_protocol="http",response_code="200",response_flags="-",source_app="sleep",source_principal="unknown",source_version="unknown",source_workload="sleep",source_workload_namespace="default"}	21`

*egress gw -> httpbin.org*
`istio_requests_total{connection_security_policy="unknown",destination_app="unknown",destination_principal="unknown",destination_service="httpbin.org",destination_service_name="httpbin.org",destination_service_namespace="default",destination_version="unknown",destination_workload="unknown",destination_workload_namespace="unknown",instance="10.56.1.11:42422",job="istio-mesh",permissive_response_code="none",permissive_response_policyid="none",reporter="source",request_protocol="http",response_code="200",response_flags="-",source_app="istio-egressgateway",source_principal="unknown",source_version="unknown",source_workload="istio-egressgateway",source_workload_namespace="istio-system"} 21`
